### ISSUE TITLE
[Question/Suggestion] When will there be a new release?

### ISSUE THREAD

**Issue by BlackLotus** [scop/bash-completion#785]
I came here to contribute pull requests for a few missing features only to find that they are already implemented.
Most distributions only package releases. Is there a specific reason why there was no release for over 2 years?
Can I do something to make a release possible?

**Comment by akinomyoga** [comment]
> Can I do something to make a release possible?

https://github.com/scop/bash-completion/projects/2

**Comment by scop** [comment]
Specifically https://github.com/scop/bash-completion/projects/2#card-80496418 is the current blocker, we need to decide and document that; after it it should be straightforward refactoring and renaming things according to the decisions.

Renaming before deciding on the calling conventions could theoretically proceed already now, but it would certainly cause breaking changes for people following the default branch and adopting the newly named functions before their API has taken its final form. Probably very few people would be affected, but I think I'd prefer to avoid that anyway.

**Comment by scop** [comment]
Closing here. I suggest using the discussions feature rather than an issue report for things like this in the future.

**Comment by dreua** [comment]
My story is exactly the same as @BlackLotus and there are probably others. I think regular releases are quite important for projects like this, otherwise people will loose interest in contributing their small improvements and need to create other solutions for this problem.
I understand and value the fact that you want to improve and refactor the underlying structure but that shouldn't hurt the project by delaying releases for too long imo. (Tried to understand and help with the open Todo but gave up. I don't have enough knowledge of the problems and probably not enough interest in the completion-internals.) Would money help to bring this forward?

**Comment by dreua** [comment]
Same in #757

**Comment by scop** [comment]
Speaking for myself, I don't quite see financial contributions making a difference on this right now.

Problems with rare releases are understood. Then again, apart from the current unfinished business/refactoring in the master branch, that branch has been in such a stable state for quite a long time that I've been semi-seriously considering making releases altogether, they're in most cases just arbitrary snapshots that just happen some random time without any particular reason. Problems arising from that are understood too, though, so I haven't gone there.

Perhaps we should just start making automated releases, let's say bi-monthly or quarter yearly ones from a "stable" branch (which is usually the main one here, currently "master").

**Comment by dreua** [comment]
It sounds to me like you are overthinking / over-engineering this. First priority should be to make a release now, any thoughts on automation can come later. 

> [Releases are] in most cases just arbitrary snapshots that just happen some random time without any particular reason

In my opinion that is absolutely fine for a project like this. Most Linux-Distros work with upstream releases so I'd recommend to stick with this. One might even argue that semver doesn't really apply here and put any small changes in a patch version update.
You could release after each little update or make a trade-off and collect changes for a few months, that's up to you. I personally would probably release every few months or whenever someone asks for it, like in this issue. (Pro: You get some social interaction out of it, i.e. the person might offer help or say thank you once it's done :wink: )

TL;DR: There is no need for exact rules here imo. Automate only if it makes your life easier. Releases at a random time are perfectly fine as long as it's not too long.

**Comment by scop** [comment]
Automation or lack of it is not in the way of a new release in any way. Comments [2](https://github.com/scop/bash-completion/issues/785#issuecomment-1206764390) and [3](https://github.com/scop/bash-completion/issues/785#issuecomment-1221233094) in this issue point the things that are. TL;DR once more: we are in the middle of refactoring and API changes, and a release will happen once that work is in a shape we are willing to maintain for the foreseeable future.
